Library/Util.ts

```diff
@@ -146,25 +146,62 @@ class Util {
     }
 
     /**
-     * Validate that an object is of type { [key: string]: string }
+     * Simplify a value into something usable for customDimensions
+     * @param value - what to inspect
+     * @param seenObjects map of objects we've already visited, for detecting circular refs
+     * @param jpath
+     * @returns may return null if we can't convert (eg; it was a function)
+     */
+    protected static toSimpleValue(value: any, seenObjects: any, jpath: string[]): string | null {
+        const valType = typeof value;
+        if (valType === "string" || valType === "number" || valType === "boolean") {
+            return value.toString().trim(0, Util.MAX_PROPERTY_LENGTH);
+        } else if (value === null || value === undefined) {
+            return "";
+        } else if (Util.isArray(value)) {
+            return JSON.stringify(value.map((arrVal: any, arrKey: any) => {
+                return Util.toSimpleValue(arrVal, seenObjects, jpath.concat(arrKey));
+            }));
+        } else if (valType === "object") {
```

Comment:
I think this will pick up Date objects as well. Is that okay?

Reply:
I'll add a test case for this